x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java

```diff
@@ -34,22 +35,44 @@
 public abstract class CommonEnrichRestTestCase extends ESRestTestCase {
 
     @After
-    public void deletePolicies() throws Exception {
-        Map<String, Object> responseMap = toMap(adminClient().performRequest(new Request("GET", "/_enrich/policy")));
-        List<Map<?, ?>> policies = unsafeGetProperty(responseMap, "policies");
-
-        for (Map<?, ?> entry : policies) {
-            Map<?, Map<String, ?>> config = unsafeGetProperty(entry, "config");
-            Map<String, ?> policy = config.values().iterator().next();
-            String endpoint = "/_enrich/policy/" + policy.get("name");
-            assertOK(client().performRequest(new Request("DELETE", endpoint)));
-
-            List<?> sourceIndices = (List<?>) policy.get("indices");
-            for (Object sourceIndex : sourceIndices) {
-                try {
-                    client().performRequest(new Request("DELETE", "/" + sourceIndex));
-                } catch (ResponseException e) {
-                    // and that is ok
+    public void deletePipelinesAndPolicies() throws Exception {
+        {
+            // delete all pipelines
+            List<String> pipelines = new ArrayList<>();
+            try {
+                Map<String, Object> responseMap = toMap(adminClient().performRequest(new Request("GET", "/_ingest/pipeline")));
```

Comment:
We're not running this in parallel with any other tests that might be creating their own pipelines are we? Is that even something to worry about here?

Reply:
That's a fair point!

My thinking here was that we're already unconditionally deleting all the enrich policies in this cleanup, so we've already got a thread-safety resource-bashing issue. And while that's true, there's probably more tests out there in the world that use ingest pipelines than tests that use enrich policies, so this makes the potential damage bigger than it was before.

I think I can limit this damage pretty well, commit coming.

Reply:
How about 9a67267f414f6023daa4f92ba4c07ce7604ecf2d, @jbaiera?